infrastructure/templates/public-api/application/public-api/publicApiApp.bicep

```diff
@@ -0,0 +1,154 @@
+@description('Specifies the location for all resources.')
+param location string
+
+@description('Specifies the Public API resource prefix')
+param publicApiResourcePrefix string
+
+@description('Specifies the Admin App Service name.')
+param adminAppName string
+
+@description('Specifies the id of the Container App Environment in which to deploy this Container App.')
+param containerAppEnvironmentId string
+
+@description('Specifies the name of the fileshare used to store Public API Data.')
+param publicApiDataFileShareName string
+
+@description('Specifies the Container Registry name from which to pull Docker images.')
+param acrName string
+
+@description('The tags of the Docker images to deploy.')
+param dockerImagesTag string
+
+@description('The URL of the Public API.')
+param publicApiUrl string
+
+@description('The URL of the Content API.')
+param contentApiUrl string
+
+@description('Specifies the Application (Client) Id of the App Registration used to represent the API Container App.')
+param apiAppRegistrationClientId string
+
+@description('Specifies the connection string template for the Public API PSQL database.')
+param publicApiDbConnectionStringTemplate string
+
+@description('Specifies a set of tags with which to tag the resource in Azure.')
+param tagValues object
+
+var apiAppName = '${publicApiResourcePrefix}-ca-api'
```

Comment:
You could consider adding an `abbreviations.json` file with the set of all abbreviations, which you can take out of one of the sample Azure template projects e.g. https://github.com/Azure-Samples/todo-csharp-sql/blob/main/infra/abbreviations.json, and then using that to reference abbreviations like `ai-`, `ca-`, `cae-`, etc, so that they are always consistent.

You would need to load it though in each file, using `var abbrs = loadJsonContent('./abbreviations.json')` , which is going to get a bit repetitive unless more names are worked out at higher level in main.bicep and passed down.

Reply:
As per the other PR - added as a new Bicep file along with types